### PR TITLE
Catch2

### PR DESCRIPTION
**Pull request by JensWehner**
Convert the testsuite to the catch2 framework. 

This does not utliize the whole power of the framework yet, as this is basically just a regex replace. 

At least for the moment I think this is a big enough improvement to merit merging into master.

### CODE REVIEW

File: tests/unit/Utilities/010BasisFunctions_UnitTest.cpp [hunk 1]
```diff
@@ -79,18 +85,17 @@ int main() {
 
             point1D[0] += -1e-8;
             ret = test->evalDeriv((point1D));
-            logger.assert_always((std::abs(ret[0] - 5.e7 * (x1 - x0)) < 1e-5),
-                                 "gradient");
-            logger.assert_always((std::abs(test->evalDeriv0((point1D)) -
-                                           5.e7 * (x1 - x0)) < 1e-5),
-                                 "derivative");
+            INFO("gradient");
+CHECK((std::abs(ret[0] - 5.e7 * (x1 - x0)) < 1e-5));
```
Comment (Laar) [inline tests/unit/Utilities/010BasisFunctions_UnitTest.cpp:89]: This seems to be badly indented. Is that a requirement that the CHECK is at the start of the line?

Comment (JensWehner) [reply tests/unit/Utilities/010BasisFunctions_UnitTest.cpp:89]: We can also require that we get the local catch2, but all the projects I have seen just put the catch header inside.

Comment (JensWehner) [reply tests/unit/Utilities/010BasisFunctions_UnitTest.cpp]: @Laar the intendaton is fixed.

**Comment by JensWehner** [comment]
@Laar please review.

File: tests/unit/Geometry/020PointReference_UnitTest.cpp [hunk 2]
```diff
@@ -1,236 +0,0 @@
-/*
```
[the remark this answers is not in the document]
Comment (JensWehner) [reply tests/unit/Geometry/020PointReference_UnitTest.cpp:1]: the whole test was commented out, before I got there. So I deleted it

[the remark this answers is not in the document]
Comment (Laar) [reply tests/unit/Geometry/020PointReference_UnitTest.cpp:1]: Hmm, I see. Strange.

**Pull request merged**